compiler/rustc_typeck/src/check/op.rs

```diff
@@ -506,7 +451,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         // This has nothing here because it means we did string
                         // concatenation (e.g., "Hello " + "World!"). This means
                         // we don't want the note in the else clause to be emitted
-                    } else if let [ty] = &visitor.0[..] {
+                    } else if lhs_ty.has_param_types_or_consts() {
```

Comment:
This is technically more general, since that visitor will fail if we have a type with two param types -- but I see no reason why that would be better.